plat/common/bootinfo.c

```diff
@@ -113,3 +119,142 @@ void ukplat_bootinfo_print(void)
 			   );
 	}
 }
+
+#ifdef CONFIG_LIBFDT
+static void fdt_bootinfo_mem_mrd(struct ukplat_bootinfo *bi, void *fdtp)
+{
+	struct ukplat_memregion_desc mrd = {0};
+	int prop_len, prop_min_len;
+	__u64 mem_base, mem_sz;
+	int naddr, nsz, nmem;
+	const __u64 *regs;
+	int rc;
+
+	/* search for assigned VM memory in DTB */
+	rc = fdt_check_header(fdtp);
+	if (unlikely(rc))
+		ukplat_bootinfo_crash("Reserved memory is not supported");
+
+	nmem = fdt_node_offset_by_prop_value(fdtp, -1, "device_type",
+					     "memory", sizeof("memory"));
+	if (unlikely(nmem < 0))
+		ukplat_bootinfo_crash("No memory found in DTB");
+
+	naddr = fdt_address_cells(fdtp, nmem);
+	if (unlikely(naddr < 0 || naddr >= FDT_MAX_NCELLS))
+		ukplat_bootinfo_crash("Could not find proper address cells!");
+
+	nsz = fdt_size_cells(fdtp, nmem);
+	if (unlikely(nsz < 0 || nsz >= FDT_MAX_NCELLS))
+		ukplat_bootinfo_crash("Could not find proper size cells!");
+
+	/*
+	 * QEMU will always provide us at least one bank of memory.
```

Comment:
This code was just taken from the place you have already pointed to, I did not bother with improving it at that time. If I were to make this suggested change, it would have been in another commit. If you really want me to, I can update it to look for other memory banks as well, it should not be difficult and I would not mind it at all :).

Reply:
Okay. Adding support for multiple banks would go together with excluding reserved memory. Perhaps to avoid delay merging this PR further we can add it later. Maybe just reduce the comment to something like "for now we only support one memory bank"?